Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -801,19 +801,23 @@ private void GroupDisplayData()
         if ( GroupBy is null )
         {
             var firstGroupableColumn = groupableColumns[0];
-            var newGroupedData = DisplayData.GroupBy( x => firstGroupableColumn.GetGroupByFunc().Invoke( x ) )
-                                                                             .Select( x => new GroupContext<TItem>( x, firstGroupableColumn.GroupTemplate ) )
-                                                                             .OrderBy( x => x.Key )
-                                                                             .ToList();
-            RecursiveGroup( 1, groupedData, newGroupedData );
+            var query = DisplayData.GroupBy( x => firstGroupableColumn.GetGroupByFunc().Invoke( x ) )
+                                   .Select( x => new GroupContext<TItem>( x, firstGroupableColumn.GroupTemplate ) );
+            var newGroupedData = ( firstGroupableColumn.CurrentSortDirection switch
+            {
+                SortDirection.Descending => query.OrderByDescending( context => context.Key ),
+                _ => query.OrderBy( context => context.Key ),

```

Comment:
I think

`_ =>` should be `SortDirection.Ascending` because we have a `Default` sort option, which should not sort by anything if sorting is not defined.